src/status_im/common/biometric/events.cljs

```diff
@@ -1,120 +1,124 @@
 (ns status-im.common.biometric.events
   (:require
-    [native-module.core :as native-module]
-    [re-frame.core :as re-frame]
     [react-native.biometrics :as biometrics]
-    [react-native.platform :as platform]
+    [schema.core :as schema]
+    [status-im.common.biometric.events-schema :as events-schema]
+    [status-im.common.biometric.utils :as utils]
     [status-im.common.keychain.events :as keychain]
     [status-im.constants :as constants]
     [taoensso.timbre :as log]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
-(def android-device-blacklisted?
-  (and platform/android? (= (:brand (native-module/get-device-model-info)) "bannedbrand")))
-
-(defn get-label-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-android (i18n/label :t/biometric-fingerprint)
-    constants/biometrics-type-face-id (i18n/label :t/biometric-faceid)
-    (i18n/label :t/biometric-touchid)))
-
-(defn get-icon-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-face-id :i/face-id
-    :i/touch-id))
-
-(re-frame/reg-fx
- :biometric/get-supported-biometric-type
+(rf/reg-fx
+ :biometric/get-supported-type
  (fn []
    ;;NOTE: if we can't save user password, we can't use biometric
    (keychain/can-save-user-password?
     (fn [can-save?]
-      (when (and can-save? (not android-device-blacklisted?))
+      (when (and can-save? (not utils/android-device-blacklisted?))
         (-> (biometrics/get-supported-type)
             (.then (fn [type]
-                     (rf/dispatch [:biometric/get-supported-biometric-type-success type])))))))))
+                     (rf/dispatch [:biometric/set-supported-type type])))))))))
 
-(rf/defn get-supported-biometric-auth-success
-  {:events [:biometric/get-supported-biometric-type-success]}
-  [{:keys [db]} supported-type]
+(defn set-supported-type
+  [{:keys [db]} [supported-type]]
   {:db (assoc db :biometric/supported-type supported-type)})
 
-(rf/defn show-message
-  {:events [:biometric/show-message]}
-  [_ error]
-  (let [code    (ex-cause error)
-        content (if (#{::biometrics/not-enrolled
+(schema/=> set-supported-type events-schema/?set-supported-type)
+(rf/reg-event-fx :biometric/set-supported-type set-supported-type)
+
+(defn show-message
+  [_ [code]]
+  (let [content (if (#{::biometrics/not-enrolled
```

Comment:
Here we're using namespaced keywords, is there any intention to refactor them to qualified keywords instead as we generally tend to do this for event names

Reply:
Not sure I get it. The keyword you highlighted is a used as an "error type". The events here still use the same qualified keywords e.g. `:biometric/authenticate`. 

I was thinking about changing them to the namespaced form e.g. `::authenticate` so they could be qualified by requiring the ns e.g. by aliasing the `biometric` ns, the event can be used as `::biometrics/authenticate`. This has some benefits, including easier go-to-definition with LSPs, but didn't see it used for events so decided to not do it.

Reply:
@clauxx, I kindly ask you to not change to namespaced keywords unless it's aligned as the new guideline. We had discussions in the past and we settled for non-namespaced keywords (in general) as that seemed to be the preference for most contributors. Consistency is our secret ingredient for productivity!

In the guidelines:

> Always register events and subscriptions using a meaningful namespace, but don't namespace them with `::`.